### ISSUE TITLE
No idea what is going on... One minute an api path works, the next i'm unauthorised...

### ISSUE THREAD

**Issue by DarceyLloyd** [getinsomnia/insomnia#4253]
### Expected Behavior

maintaining session and not getting access denied 401

### Actual Behavior

So the web application is 90% built and has been fine for months, but now all of a sudden I lose authentication and session on a once working and untouched api route. If I rename the api route to abc it works again but if I rename it back, it doesn't work again...

Tested this in browser and postman and was fine in both...

http://127.0.0.1/api/advocate/cases (access denied)
http://127.0.0.1/api/advocateeeee/cases (works)
http://127.0.0.1/api/advisors/cases (works - same class and method different route)

I can't supply export as the project is under NDA and has sensitive material.

Uninstalled, went to appData folder, removed all the left behind stuff (nice un-install guys!) and then re-installed. Re-imported exported data and all is working again...

### Reproduction Steps

NDA can't share files.

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

2021.6.0

### What operating system are you using?

Windows

### Operating System Version

10 pro latest

### Installation method

download from website

### Last Known Working Insomnia version

2021.4.1

**Comment by beingflo** [comment]
Not sure if it's the same issue but I also just wasted an hour with suddenly getting 401s even though nothing changed. For me the problem was that the cookie I'm getting from the `/session` path is not being sent to other routes anymore. The cookies has `Path=/session` set even though I don't do that. If I manually remove that it works as expected.
Note sure if that's a regression from insomnia.

**Comment by DarceyLloyd** [comment]
Yes, I have to delete all cookies every 15 to 30 minutes or so or certain routes just stop working... Saved me from using postman, which I can't stand. I would rather just use browser dev tools than that...

Pretty major bug...

**Comment by filfreire** [comment]
@DarceyLloyd thanks for reporting this issue!

We could use some more information to try to debug this.
- Does this still happen on latest `2021.7.2`
- Are you using any cache control headers? (e.g. `Cache-Control: no-cache`)


@beingflo - in your case, are you still having issues also past version `2021.7.2`?

**Comment by DarceyLloyd** [comment]
RE: Does it still happen on 2021.7.2
Yep, was using the application this morning, had to dump all cookies before it would work again, random routes just starting to stop working... It sometimes takes a good 30 minutes to an hour of use before it happens or it can start happening fairly quickly... No issues with postman or ARC, however I hate postman and ARC needs work.

RE: Headers
LIVE: header("Access-Control-Allow-Origin: " . Config::$domain);
DEV: header("Access-Control-Allow-Origin: *");
header('Content-type:application/json;charset=utf-8');
header('Cache-Control: no-cache, no-store');

**Comment by beingflo** [comment]
@filfreire Seems to work now, thanks!

**Comment by warrenc5** [comment]
I had this and cookies were getting corrupted.

This is correct cookie I pasted from raw

![shot-2022-11-10_08-29-59](https://user-images.githubusercontent.com/3395365/200963403-6c5dd467-5bd9-4094-93e4-6374c17d9070.jpg)

This is corrupted cookie after first post.
![shot-2022-11-10_08-25-20](https://user-images.githubusercontent.com/3395365/200963408-ac7d3d4d-548e-4c8e-9099-3a56898905df.jpg)

The original cookie was 

`beeline_tfnsw_microFrontendAuth0TokenInfo=BENCcAJr3/el2F0WTX9Nlbtmjd35i1fxQfVw9YQ2SjnoCbUp9BIBPMfHFoCO4+qZ2vccEDDiBIPG38R98i73/RnG5rbwqO31VEhEPgb5npn88hsdPgkQkImM9W794TTUUSZ5MlwNIArhEwfbQtnEFlG5xlF1E4VcXZ0lbH0qlET/yLsj1iVjRtLSOke2Z+bunmmF92RjCyQPp+XsvFwBjzCZG9vVDvZIP5NCXMqDh8jH8E0tzwEzMIN84IHY/VWsvOmrTRQaJPVjMmyFGRT59Sk6THD4ba3JwWFBJWjCOcGQD0QiKus3zD/+zp9v9otIkg9vSCnumBpneYDKz+VKBFygjj92mB+qmljWDhWkaWHUHoIJ1F/xdmouEEHjZOI1dY71X6Tfx1WePCtec5hyJzPUM1SW3BXSLqMeKHO1JTWcjJzvMD+k31iZ+0hJumo9XRNcDwNLX1cY6qkCE5UVRXJgRxUAe1VvWFTbQgeQ33C8vvyEzYXonWaGwMT4K8YMLY/JlRpq4CkjtpdMqLmqdNE7AS+mrtfa1GVkepVa8c6Ow9DJ0HeibPlbQOfO55ypRDj8hM2F6J1mhsDE+CvGDC2PycR2ELp1F5H+oo4cYqTvaBfApmqoWn6bbbyommLTmhZnxAYxVdXbjBdm3nysdVfhf/dkYwskD6fl7LxcAY8wmRvb1Q72SD+TQlzKg4fIx/BN0j6IVmiS0xNVYy3ZpZJm0fc/upEIr7NCkzMJgxgylLcUo7fpaP7hp7pR/uvncjZ5u2gc0V/ZWB8ZYAokvFRo0WGRB0UdeZtp3zxi7SOgUlCRRkdfKvpTV6FfRRU/jZhOGelb46myHotSbP3vNyKK0XEwqvN/s5qRKmBfdu5FWCLqlnYQB82KF2ZVFtM2K/7UB0ub+8NRh7Fu1Rj/sV7an0KedKax6ICDO9S2JMklZll5U65atIwamCyZ30gyxFdwD+LXpwlN+1ke2hAJdyeSSe3swEVyv2/V9p+OOlpMnRPgBbHtcZf//JY+2mhMwXENGbnMZMWfCrgVJlYAedDtlbMSeC5tY/VKdmivVcNZr8JJDTHTeZ+5yeX1lD3PQOVdgHQ4I+yoYPSBi7viQgj+GSBlKIKRfR+TPHLXfrVWf127PY8WugT68StP/JN3oNRNNxXCrdyk58v5kWlsAvk7Z0OPTKFy2Im3moyAzhkeg7fHddX5xNus3N+PVwUL24RafXk1lNlfdynA9YWsmoHX4sq+vHECNqSM5F3WwLyO50fxXtrEVlbk59FNJTURUrxYO43hAGj+vFVglhIz3WQ4ddwHi18vGprBJsHl92R0hv2bVvCxqDJKjgut6UKDoyi6c5UuqepmbNripwvfk55Mg815LVwv/u6g0xU2Dp2HxghrM+5Bnvuz2VvIJf8UR26PYtJfkcAdaDmDpX5SGTs7UNO2e2VQuPkNHDRfJztmjbG4/IQy6/UOVb+LdCogMRf3MENKS5b7fewHkZTkslQHFQiKpL2Itk8ct0itHqQR2K7lK1MsDSmXHYQdRlm9/xylBFf2IYdS6hiuY+awUmZvcssGMU4x5zRwaKgG4kg5XCnDgJUMFM/ErzAgLdWZtEfRHWZ74NJsv0DUtoTDEkt0d4MbhNGbyx/Q3Y/OVyUpZz3XO+PT7g99m85QDa8PBmOthyKBWfTz/dYgz3bv6ZxKEh97Q/GF4c+IcYRccxDT097bkuaDBXU0tev7QXlkqS7h7XaVCFRSKhQ9zRkYdNYb8k+Qnp0E3LM64kOkdrsugNOl/gHOCoySH6Okw4713gPPdYoTh2vWJtqV/2RYBBPcq/vPv3UxdmXl/fiV93qb1w5/ctKmcIKTmCrOlxVwk8Z9vmJ67VAsH2//IxwTr14TX9qi0lljg6mNVbMWi9E0gOM=; ASP.NET_SessionId=zn130papjbhylrjfaifhyvbx; AntiXsrfToken=1535c5689dfd4607b807ae93dbc50621; BWVSID=ba48b91b-1126-406f-89e2-ffb7c24ba3fe; bc=%7B%22isUrlRewritten%22%3Afalse%2C%22title%22%3A%22My%20Contact%20Information%22%2C%22url%22%3A%22%2Ftfnsw%2FHome%2FContactInformationScreen.ascx%22%7D%7C%7B%22isUrlRewritten%22%3Afalse%2C%22title%22%3A%22Preferences%20-%20My%20Account%22%2C%22url%22%3A%22%2Ftfnsw%2FHome%2FUserPreferencesScreen.ascx%22%7D; a73sm3=prodweb108.jax; __cfruid=5d178dbed1dc05e3c8e69610124a19d519765f03-1668025744; .BEEAUTH_tfnsw=76b2e81f-ea9b-46e0-abe9-d04d8f92ef02`
